### ISSUE TITLE
SchismTracker still segfaults on GNU/Linux with ALSA SDL2 driver

### ISSUE THREAD

**Issue by vanfanel** [schismtracker/schismtracker#321]
Hi there,

On my GNU/Linux systems, Schismtracker just plain segfaults. It happens on Debian and Raspberry Pi OS (which is based on Debian too), where I only have ALSA, and I don't have or want legacy OSS at all.

Some modern systems like Raspberry Pi OS lack OSS/DSP, so SDL2 uses ALSA audio.

Schismtracker does some VERY strange things to open the audio device in `_audio_open()` here:

-First it enforces OSS stuff here if no driver in `~/.schism/config` or `SDL_AUDIODRIVER` is set:
https://github.com/schismtracker/schismtracker/blob/master/schism/audio_playback.c#L1290
Please don't do that. Most of us don't care/like/have OSS compatibility modules. OSS has been dead for 20 years. Let it rest in peace. An HW ALSA device is the way to go to get low-latency audio.

-Then it adds **"hw"** to the driver name if the driver is **alsa**, here:
https://github.com/schismtracker/schismtracker/blob/50a7a3e493070f54d5dc10136cd38b8ea9d6f9db/schism/audio_playback.c#L1302
That makes zero sense. Don't add anything to the driver name. How do you know I have `alsa:hw` as an usable device? That's only working for you, in your particular system.

That said, I can't get Schismtracker to work on GNU/Linux unless I remove that code.
I can do debugging and building, but I am afraid many users will approach Schismtracker on GNU/Linux and simply run away from it because all it does, as things are, is segfault badly.

In a nutshell: Stop forcing driver names, etc. Use whatever SDL2 uses by default, please. What you are doing right now does not work AT ALL.

**Comment by vanfanel** [comment]
This is not a legit issue. The cause for the segfaut is different. Closing this.
